cmd/zc_enumerator.go

```diff
@@ -255,6 +279,12 @@ func initResourceTraverser(source string, location common.Location, ctx *context
 			return nil, err
 		}
 
+		if sourceURL.Scheme == "http" {
+			httpsRecommendationOnce.Do(func() {
+				glcm.Info(httpsRecommendedNotice)
+			})
+		}
```

Comment:
can we perhaps extract out the part of the code where the URL is parsed and "http" is checked?

Reply:
so that it's not repeated

---
In reply to: [321544752](https://github.com/Azure/azure-storage-azcopy/pull/577#discussion_r321544752) [](ancestors = 321544752)

Reply:
Oh, I just letf a comment to the same effect :-)  Will leave my comment in too.

Reply:
Done